connectors/src/connectors/intercom/temporal/sync_help_center.ts

```diff
@@ -0,0 +1,383 @@
+        deleteFromDataSource(
+          {
+            dataSourceName: dataSourceConfig.dataSourceName,
+            workspaceId: dataSourceConfig.workspaceId,
+            workspaceAPIKey: dataSourceConfig.workspaceAPIKey,
+          },
+          dsArticleId
+        ),
+        article.destroy(),
+      ]);
+    })
+  );
+
+  // Then we delete the collection
+  const dsCollectionId = getHelpCenterCollectionDocumentId(
+    collection.intercomWorkspaceId,
+    collectionId
+  );
+  await Promise.all([
+    deleteFromDataSource(
+      {
+        dataSourceName: dataSourceConfig.dataSourceName,
+        workspaceId: dataSourceConfig.workspaceId,
+        workspaceAPIKey: dataSourceConfig.workspaceAPIKey,
+      },
+      dsCollectionId
+    ),
+    collection.destroy(),
+  ]);
+  logger.info(
+    { ...loggerArgs, collectionId },
+    "[Intercom] Collection deleted."
+  );
+
+  // Then we call ourself recursively on the children collections
+  const childrenCollections = await IntercomCollection.findAll({
+    where: {
+      connectorId,
+      parentId: collectionId,
+    },
+  });
+  await Promise.all(
+    childrenCollections.map(async (collection) => {
+      await _deleteCollection({
+        connectorId,
+        dataSourceConfig,
+        collection,
+        loggerArgs,
+      });
+    })
+  );
+}
+
+/**
+ * Syncs a collection and its children (collection & articles) from the database and the core data source.
+ */
+async function _syncCollection({
+  connectorId,
+  intercomClient,
+  dataSourceConfig,
+  loggerArgs,
+  collection,
+  parents,
+}: {
+  connectorId: ModelId;
+  intercomClient: IntercomClient;
+  dataSourceConfig: DataSourceConfig;
+  collection: IntercomCollectionType;
+  loggerArgs: Record<string, string | number>;
+  parents: string[];
+}) {
+  // Sync the Collection
+  let collectionOnDb = await IntercomCollection.findOne({
+    where: {
+      connectorId,
+      collectionId: collection.id,
+    },
+  });
+
+  if (collectionOnDb) {
+    await collectionOnDb.update({
+      name: collection.name,
+      description: collection.description,
+    });
+  } else {
+    collectionOnDb = await IntercomCollection.create({
+      connectorId: connectorId,
+      collectionId: collection.id,
+      intercomWorkspaceId: collection.workspace_id,
+      helpCenterId: collection.help_center_id,
+      parentId: collection.parent_id,
+      name: collection.name,
+      description: collection.description,
+      url: collection.url,
+      permission: "read",
+    });
+  }
+
+  const collectionContent = await renderDocumentTitleAndContent({
+    dataSourceConfig,
+    title: collection.name,
+    content: { prefix: null, content: collection.description, sections: [] },
+    createdAt: new Date(collection.created_at),
+    updatedAt: new Date(collection.updated_at),
+  });
+  const collectionDsDocumentId = getHelpCenterCollectionDocumentId(
+    collection.workspace_id,
+    collection.id
+  );
+  await upsertToDatasource({
```

Comment:
This document is basically empty no?

Reply:
Yes Collections only store a description. I considered 3 options: 
1/ Adding them as datasource (this PR)
2/  Not adding them as datasource
3/ Not adding them as datasource but adding the description of the parent collections on top of the article content. 

Do you recommend switching to option 2/ or 3/ ?

Reply:
I would go with 3 but not as prefix as I presume they can be long. Just as intro of each document?
The collection name can be used as prefix I presume?

So I would repeat article title and collection name everywhere and inject the collection description at the top of the document which will serve a purpose for small-ish document and otherwise will be a bit repetitive... but fine.

Reply:
awesome, thanks 🙏🏻